unit/test_edit_topology.py

```diff
@@ -0,0 +1,589 @@
+        self.instance.add_unjoined_server(alias='r1-replica', uri='r1-replica-uri')
+        self.instance.add_unjoined_server(alias='r1-replica-2', uri='r1-replica-2-uri')
+        self.instance.add_unjoined_server(alias='r2-leader', uri='r2-leader-uri')
+        # now we don't don't add r2-replica
+
+        # create replicasets with instances not known by cluster (r2-replica)
+        self.instance.clear_calls('edit_topology')
+        res = call_edit_topology(
+            self.console_sock,
+            hostvars
+        )
+        self.assertFalse(res.success, msg=res.msg)
+        self.assertIn("Some of replicaset instances aren't found in cluster: r2-replica", res.msg)
+        self.assertEqual(len(self.instance.get_calls('edit_topology')), 0)
+
+        # add r2-replica
+        self.instance.add_unjoined_server(alias='r2-replica', uri='r2-replica-uri')
+
+        # create replicasets
+        self.instance.clear_calls('edit_topology')
+        res = call_edit_topology(
+            self.console_sock,
+            hostvars
+        )
+        self.assertTrue(res.success, msg=res.msg)
+        self.assertTrue(res.changed)
+
+        calls = self.instance.get_calls('edit_topology')
+        self.assertEqual(len(calls), 1)
+
+        call = calls[0]
+        self.assertNotIn('servers', call)
+        self.assertIn('replicasets', call)
+
+        replicasets_opts = call['replicasets']
+        self.assertEqual(len(replicasets_opts), 2)
+
+        exp_replicasets_opts = [
+            {
+                'alias': 'r1',
+                'roles': ['role-1', 'role-2'],
+                'all_rw': True,
+                'weight': 1,
+                'vshard_group': 'group-1',
+                # no failover priority opt, just join_servers in right order
+                'join_servers': [
+                    {'uri': uri}
+                    for uri in ['r1-leader-uri', 'r1-replica-uri', 'r1-replica-2-uri']
+                ]
+            },
+            {
+                'alias': 'r2',
+                'roles': ['role-2'],
+                # no failover priority opt, just join_servers in right order
+                'join_servers': [
+                    {'uri': uri}
+                    for uri in ['r2-leader-uri', 'r2-replica-uri']
+                ]
+            }
+        ]
+
+        self.assertEqual(replicasets_opts, exp_replicasets_opts)
+
+    def test_change_replicasets(self):
+        rpl1_vars = {
+            'replicaset_alias': 'r1',
+            'roles': ['role-1', 'role-2'],
+            'failover_priority': ['r1-leader', 'r1-replica'],
+            'all_rw': True,
+            'weight': 1,
+            'vshard_group': 'group-1'
+        }
+
+        add_replicaset(
+            self.instance, rpl1_vars['replicaset_alias'], rpl1_vars['roles'],
+            ['r1-leader', 'r1-replica', 'r1-replica-2'],
+            all_rw=rpl1_vars['all_rw'], weight=rpl1_vars['weight'],
+            vshard_group=rpl1_vars['vshard_group']
+        )
```

Comment:
Let's use keys for all args, it's more readble:
```suggestion
        add_replicaset(
            instance=self.instance,
            alias=rpl1_vars['replicaset_alias'], 
            roles=rpl1_vars['roles'],
            servers=['r1-leader', 'r1-replica', 'r1-replica-2'],
            all_rw=rpl1_vars['all_rw'],
            weight=rpl1_vars['weight'],
            vshard_group=rpl1_vars['vshard_group']
        )
```